### ISSUE TITLE
Crashing whenever crafting and item ID's -1

### ISSUE THREAD

**Issue by donovanvaught** [jaquadro/GardenCollection#41]
as the title says. I'm trying to run a server to play at home with the wife. loads up fine with the mod installed on my custom modpack but as soon as I try to craft anything it crashes on picking up the item from the crafting table. As an aside, I noticed in NEI that all (or most) of the items have an ID of -1 or -1:(some other number).  I believe I have posted the relevant crash report. If not let me know and I will replicate the problem for another report. Thank you for any help.
[crash-2015-11-30_00.19.52-client.txt](https://github.com/jaquadro/GardenCollection/files/47108/crash-2015-11-30_00.19.52-client.txt)

**Comment by jaquadro** [comment]
Sorry, I can't see anything in your crash report that implicates Garden Stuff as the source of the crash.  It sounds like there might be deeper issues with your current world

**Comment by donovanvaught** [comment]
thank you for responding, it's possible I included the wrong crash report. As replicating the error ruins my player inventory. So it's a pain to do. I'll run over and reproduce the crash for you now. I'm fairly certain it's at least something to do with Gardenstuff as it only crashes when crafting items from the mod. I'll be right back please don't go anywhere! :)

**Comment by donovanvaught** [comment]
on further inspection it looks like it has something to do with running the mod on a multiplayer server specifically. Here is the latest client crash report. I'd also like to state that attempting to have the mod on the server causes the server to fail to start. I'm including the report for the server as well. any help you have on getting the mod to work on the server is appreciated.
[crash-2015-12-05_16.11.50-server.txt](https://github.com/jaquadro/GardenCollection/files/52997/crash-2015-12-05_16.11.50-server.txt)
[crash-2015-12-05_16.09.24-client.txt](https://github.com/jaquadro/GardenCollection/files/52999/crash-2015-12-05_16.09.24-client.txt)

**Comment by jaquadro** [comment]
So one thing that sticks out is your version of Forge is very old.  Forge manages the low level ID mapping, which is where things are going wrong.  I know early versions of Forge had bugs managing those mappings.

I'd suggest making a complete backup of your world directory, then updating Forge to the latest 1.7.10 version.  Since that's jumping forward by 15 months, there's probably other mods in your pack that you'll need to update too (anything that adds fluids, for example).

I can't guarantee that will get things working, because you may already have bad data embedded in your world.  I don't know if Forge will clean it up or not.  You'll have to try it to know.

**Comment by donovanvaught** [comment]
Thank you, unfortunately trying to update forge on the server side causes the server to not start up properly.  Any advice you have there or on cleaning up the "embedded bad data" would be welcome. Aside from that I assume I just have to meticulously go through and update every single mod along with forge?

**Comment by jaquadro** [comment]
I'm not sure what "not start up properly" means, but it could be out of date mods.  If you get past that, Forge may still clean your registry.  I don't know.

@codewarrior0 Does MCedit have any sanitization capabilities on the Forge block/item registries?

**Comment by donovanvaught** [comment]
By that I meant that the server crashes instead of starting up all the way. When I get home after work I can try to update forage on the server and begin producing crash reports on that if you need me too but I don't expect you to help me with a problem that's not specifically related to your mod. :)

**Comment by codewarrior0** [comment]
> Does MCedit have any sanitization capabilities on the Forge block/item registries?

Nope. I assume Forge will remove blocks that aren't in the ID registry. I remember a case where it wouldn't remove an item for a mod that was removed, because the item was referenced from an HQM quest file. I imagine there are some other corner cases where it can't remove an invalid item, like custom inventories or extra entity tags, etc. I'm not sure what it does about enchantments.

**Comment by donovanvaught** [comment]
just popping by to say thank you again so much for the help. I have the app working on both server and client with the latest forge on server but an older forge on client. Client still crashes if I skip too far ahead on the forge side, so I assume it's in need of mod updates as well as updating one forge version at a time till we get to the latest. I'll continue to mess with this but wanted to say thank you and that you can close this issue if you like.
